### DISCUSSION TITLE
Change from 1.0 to 1.2 Broke Liquid Rendering with Bag Containing Bag

### DISCUSSION THREAD

**Discussion by remesq** [OrchardCMS/OrchardCore#11069] [Q&A]
I'm not entirely sure what I am getting wrong. I have looked at a lot of issues, read the docs, etc.

In 1.0 I created a Content Type called `GroupParagrah` that contained a Named Bag Part called `ParagraphBag`, which had a Content Type called `GroupParagraphElement` selected. This was meant to be used in separate Named Bag Parts within other Content Types so I could reuse it.

I created a `SiteFooterInformation` Content Type which is stereotyped as a Widget. It contains a Named Bag Part called `FooterInformationBag`, which has the `GroupPararaph` Content Type selected.

My code worked fine in 1.0. All Content Types were created in the Admin and the following Liquid files were in my Theme's `~/Views` folder:

`Widget-SiteFooterInformation.liquid` code:

```
    {% assign item = Model.ContentItem.Content.FooterInformationBag.ContentItems %}
    <div class="col-xs-12 col-sm-6 col-md-3 footer-entry">
        {% for paragraphItem in item %}
        {% if paragraphItem.GroupParagraph %}
            {{ paragraphItem | shape_build_display: "Detail" | shape_render }}
        {% endif %}
        {% endfor %}
    </div>
```

`GroupParagraph-ParagraphBag.liquid` code:

```
    {% if paragraphItem.GroupParagraph.ParagraphLead.Text %}
    <div class="{{ paragraphItem.GroupParagraph.ParagraphLeadClass.Text }}">
        {{ paragraphItem.GroupParagraph.ParagraphLead.Text }}
    </div>
    {% endif %}

    <div class="{{ paragraphItem.GroupParagraph.ParagraphCollectionClass.Text }}">
    {% for paragraphElement in Model.ContentItems %}
        ///// The following is the only thing that renders with OC 1.2
        {{ paragraphElement.Content.GroupParagraphElement.ParagraphElementItem.Html | raw }}
    {% endfor %}
    </div>
```

I couldn't figure out what has changed from 1.0 to 1.2 that would cause this to not render any more.  The only thing that renders is `ParagraphElementItem`. Everything else will not render. It shows up on the Console Log. And I used a `placement.json` before with a differentiator so that the elements themselves would not just be dumped in the display (previously, with 1.0, I had to set the differentiator to `"place": "-"` so that the display dump would not occur), but otherwise it worked as I have the code above.

I tried a lot of different things. Any help is appreciated.

**Comment by deanmarcussen** [comment]
Can you have a read of this issue, where some of the details of what has changed with the liquid syntax (making it more exact to the specifications)

https://github.com/OrchardCMS/OrchardCore/issues/10991

**Reply by remesq** [reply]
I saw this issue but didn't try `nil`. However, as it relates to @Scrypt suggestion notice that the other `{{ paragraphItem.GroupParagraph.ParagraphCollectionClass.Text }}` is not within an `if` condition and is still not displaying.

I am thinking that the bag within a bag has been affected by the Fluid changes. Note also that the `GroupParagrpah` items are in the top-level bag. This worked before with OC 1.0. The `GroupParagraphElement` is the second-level bag and that is being displayed in this template. 

I might have to tinker with putting this back in the top-level bag template, however, it might not work the way I have it set up.

Thanks. Will write back with my results later tonight as I am at work now.

**Comment by Skrypt** [comment]
Try something like this :

```
{% if paragraphItem.GroupParagraph != nil %}
```

```
{% if paragraphItem.GroupParagraph.ParagraphLead.Text != nil %}
``` 

I think the issue might be that your "if" conditions are not returning boolean values.

**Reply by remesq** [reply]
The other `{{ paragraphItem.GroupParagraph.ParagraphCollectionClass.Text }}` is not within an `if` condition, and still does not display. See my comment to @deanmarcussen above.

Also, I tried taking those statements out of the `if` condition and it still didn't work.

Thank you.

**Comment by Skrypt** [comment]
Ok so basically you think that the `{{ paragraphItem | shape_build_display: "Detail" | shape_render }}` fails to parse the Liquid file appropriately. Would need to debug with source code and see what happens exactly.

**Reply by remesq** [reply]
I'll have a go at that later tonight. Just a few notes:

- `ConsoleLog` on the displayed page contains all the information of the Content Items. So what I am expecting to display does appear in the JSON.
- I used a `placement.json` file with a differentiator set to `"-"`. This was done because the text of each field would display surrounded by a `<div class="[name of type and field]">[Text displays from field]</div>`. It would not display where I wanted/expected, but apart. If I remove those differentiators from the `placement.json` I get the same behavior. So the information is there, it is being passed, just not displayed where I want (in some cases as a `css` class, in other cases just as plain text).
